app/src/main/java/org/wikipedia/createaccount/CreateAccountActivity.kt

```diff
@@ -126,49 +126,53 @@ class CreateAccountActivity : BaseActivity() {
-                        if (token.isNullOrEmpty()) {
-                            handleAccountCreationError(getString(R.string.create_account_generic_error))
-                        } else if (!captchaId.isNullOrEmpty()) {
-                            captchaHandler.handleCaptcha(token, CaptchaResult(captchaId))
-                        } else {
-                            doCreateAccount(token)
-                        }
-                    }) { caught ->
-                        showError(caught)
-                        L.e(caught)
-                    })
+            lifecycleScope.launch {
+                try {
+                    val response = withContext(Dispatchers.IO) {
+                        ServiceFactory.get(wiki).getAuthManagerInfo()
+                    }
+                    val token = response.query?.createAccountToken()
+                    val captchaId = response.query?.captchaId()
+                    if (token.isNullOrEmpty()) {
+                        handleAccountCreationError(getString(R.string.create_account_generic_error))
+                    } else if (!captchaId.isNullOrEmpty()) {
+                        captchaHandler.handleCaptcha(token, CaptchaResult(captchaId))
+                    } else {
+                        doCreateAccount(token)
+                    }
+                } catch (caught: Exception) {
+                    showError(caught)
+                    L.e(caught)
+                }
+            }
         }
 
     private fun doCreateAccount(token: String) {
         showProgressBar(true)
         val email = getText(binding.createAccountEmail).ifEmpty { null }
         val password = getText(binding.createAccountPasswordInput)
         val repeat = getText(binding.createAccountPasswordRepeat)
-        disposables.add(ServiceFactory.get(wiki).postCreateAccount(getText(binding.createAccountUsername), password, repeat, token, Service.WIKIPEDIA_URL,
-                email,
-                if (captchaHandler.isActive) captchaHandler.captchaId() else "null",
-                if (captchaHandler.isActive) captchaHandler.captchaWord() else "null")
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe({ response ->
-                    if ("PASS" == response.status) {
-                        finishWithUserResult(response.user)
-                    } else {
-                        createAccountEvent.logError(StringUtil.removeStyleTags(response.message))
-                        throw CreateAccountException(StringUtil.removeStyleTags(response.message))
-                    }
-                }) { caught ->
-                    L.e(caught.toString())
-                    createAccountEvent.logError(caught.toString())
-                    showProgressBar(false)
-                    showError(caught)
-                })
+        lifecycleScope.launch {
```

Comment:
Would it be possible to use `CoroutineExceptionHandler` instead of a `try catch`?

Reply:
Applied in cde801e500c1b1ce72193e19273fc749cc299b62